analyzers/src/SonarAnalyzer.VisualBasic/SymbolicExecution/Roslyn/CalculationsShouldNotOverflow.cs

```diff
@@ -29,6 +29,28 @@ public sealed class CalculationsShouldNotOverflow : CalculationsShouldNotOverflo
 
     public override bool ShouldExecute()
     {
-        return true;
+        var walker = new SyntaxKindWalker();
+        walker.SafeVisit(Node);
+        return walker.Result;
+    }
+
+    private sealed class SyntaxKindWalker : SafeVisualBasicSyntaxWalker
+    {
+        public bool Result { get; private set; }
+
+        public override void Visit(SyntaxNode node)
+        {
+            if (!Result)
+            {
+                Result = node.IsAnyKind(
+                    SyntaxKind.AddExpression,
+                    SyntaxKind.AddAssignmentStatement,
+                    SyntaxKind.MultiplyExpression,
+                    SyntaxKind.MultiplyAssignmentStatement,
+                    SyntaxKind.SubtractExpression,
+                    SyntaxKind.SubtractAssignmentStatement);
```

Comment:
```suggestion
                Result = node.IsAnyKind(
                    SyntaxKind.AddExpression,
                    SyntaxKind.AddAssignmentStatement,
                    SyntaxKind.SubtractExpression,
                    SyntaxKind.SubtractAssignmentStatement,
                    SyntaxKind.MultiplyExpression,
                    SyntaxKind.MultiplyAssignmentStatement,
                    SyntaxKind.ExponentiateExpression,
                    SyntaxKind.ExponentiateAssignmentStatement);
```

Reply:
`a ^ b` returns `Double` and cannot overflow => no need to run

Reply:
Should we update the rule implementation then?

Reply:
Hmm, yes, indeed.